service_provider.go

```diff
@@ -1545,8 +1547,10 @@ func (sp *ServiceProvider) ValidateLogoutResponseRedirect(query url.Values) erro
 	}
 
 	if err := sp.validateSignature(doc.Root()); err != nil {
-		retErr.PrivateErr = err
-		return retErr
+		if err != errSignatureElementNotPresent && !hasValidSignature {
```

Comment:
Probably, we should check signature requirement. If everything is empty, then it's possible to just bypass verification.